keanu-project/src/main/java/io/improbable/keanu/tensor/NumberTensor.java

```diff
@@ -29,6 +29,39 @@
     
     T abs();
 
+    /**
+     * Find the index into the flattened array of the tensor of the largest value, e.g.
+     * <pre>
+     * DoubleTensor tensor = DoubleTensor.arange(0, 6).reshape(2, 3);
+     * // [[0., 1., 2.],
+     * //  [3., 4., 5.]]
+     * IntegerTensor max = tensor.argMax();
+     * // [[5]]
+     * </pre>
+     *
+     * @return A scalar tensor with the index of the largest value
+     */
+    IntegerTensor argMax();
```

Comment:
Why doesn't this just return an `int`?

Reply:
What happens if there's two values that are the max? are we meant to return both indices?

eg:
```
DoubleTensor tensor = DoubleTensor.create(1, 2, 4, 3, 4, 1, 3, 1).reshape(2, 4);
tensor.argMax();
```

Which index of `4` do we return?

Reply:
This will return the first instance of 4, (so `2`). There are tests for this but I'll also document this.

Good call on returning an `int`, I'll get that in